### ISSUE TITLE
Cannot scroll after reloading page in 8.0.1

### ISSUE THREAD

**Issue by Sabir001** [reactstrap/reactstrap#1661]
- reactstrap version `#8.0.1`
- react version `#16.10.2`
- bootstrap version `#4.3.1`

### Problem:
Yesterday I upgraded my reactstrap version to 8.0.1. But, I faced a bug. If I reload the page, the scrollbar gets removed and I cannot scroll anymore on that page. 

I also observed that this only happens in pages in which generally there is no pre-populated data. Data is fetched by API calls and then shown on the page. 

I downgraded reactstrap to 7.1.0. And everything is now working perfectly.

**Comment by Sabir001** [comment]
> Are there any error in the devTools console?
> The only thing I can think of which could cause this is the `.modal-body` class being applied to the `<body>`. Can you check if that class is being applied?

No error in devtools.
Yes.  `.modal-body` is being appended in 8.0.1.
It does not get appended in 7.0.1

**Comment by koolamusic** [comment]
Can I take this, any help to file paths that I should keep in mind ?

**Comment by TheSharpieOne** [comment]
Sure @koolamusic. It's most likely coming from the https://github.com/reactstrap/reactstrap/blob/master/src/Modal.js triggered by mounting or something that ends up calling the `.init()` which handles conditionally adding the class to the body:
https://github.com/reactstrap/reactstrap/blob/47843998b627aadcccfe71b092a194197fd2ce15/src/Modal.js#L271-L278
which is later supposed to be cleaned up in the `.close()` which conditionally removes that class
https://github.com/reactstrap/reactstrap/blob/47843998b627aadcccfe71b092a194197fd2ce15/src/Modal.js#L299-L306

Something seems to be triggering it when it isn't suppose to or maybe not cleaning it up when it is suppose to.

**Comment by koolamusic** [comment]
I ran some tests and not able to replicate this bug

**Comment by TheSharpieOne** [comment]
@Sabir001 is there any other information you can provide (maybe a [stackblitz](https://stackblitz.com/edit/reactstrap-v8?file=Example.js) demo)?

**Comment by Sabir001** [comment]
The project I observed the bug, has too many packages connected. I could not reproduce in Stackblitz. I will that try to dig the reason. If I can find, I will update here ...

**Comment by TheSharpieOne** [comment]
I'm guessing this is more directly related to #1626
Not sure if you can confirm or not, but does your project initialize a modal with `isOpen` set to `true` and hides it via other conditionals? (similar to the example provided in #1626).

**Comment by Sabir001** [comment]
It is midnight in my Timezone. I'll test tomorrow and update here.

**Comment by Sabir001** [comment]
> I'm guessing this is more directly related to #1626
> Not sure if you can confirm or not, but does your project initialize a modal with `isOpen` set to `true` and hides it via other conditionals? (similar to the example provided in #1626).
> 
> as a test, can you import Modal at your project root (such as your `index.js` or `app.js` and add this to see if it affects it
> 
> ```js-jsx
> Modal.prototype.componentWillUnmount = function () {
>   if (this.props.onExit) {
>     this.props.onExit();
>   }
> 
>   if (this._element) {
>     this.destroy();
>     if (this.props.isOpen) {
>       this.close();
>     }
>   }
> 
>   this._isMounted = false;
> }
> ```

@TheSharpieOne, even after adding the snippet in `index.js`, the problem still persists. 

Also, I wanted to say this again, 
> I also observed that this only happens in pages in which generally there is no pre-populated data. Data is fetched by API calls and then shown on the page.

**Comment by Sabir001** [comment]
I think the issue is connected to #1626 ...
`modal-open` persisted in `body` tag in my case, too.

**Comment by TheSharpieOne** [comment]
Yeah, I thought it might be the same as #1626, which is why I wanted you to see that modification. Since it didn't work, it might not be related. Here is the example from #1626 with that modification: https://codesandbox.io/s/buggy-modal-e7trx You can see it enable scrolling / fixes the issue.

**Comment by 2021eo3ar** [comment]
@TheSharpieOne could you please assign this issue to me, I can comp up with a better solution
